### ISSUE TITLE
Files written to standard XDG dirs even when run in a basedir

### ISSUE THREAD

**Issue by toofar** [qutebrowser/qutebrowser#7978]
**Version info**:
qutebrowser v3.0.1
Git commit: 25e0d02de4b2 on pakjoy (2023-10-28 16:15:45 +1300)
Backend: QtWebEngine 6.6, based on Chromium 112.0.5615.213 (from api)
Qt: 6.6.0

**Does the bug happen if you start with `--temp-basedir`?**:
Good question! Yes it does.

**Description**
When running qutebrowser with the `--basedir` argument I would like it to not write to any files outside of that basedir (beyond /tmp/ and /dev/shm/ I suppose). It looks to me like various cache files are written to the standard XDG dirs in $HOME even when they should be written to basedir/cache/.

I'm not sure if my expectations match other peoples or not. Is this a bug? Or is it normal?

**How to reproduce**

```bash
mkdir /tmp/basedir-isolation
cd /tmp/basedir-isolation
mkdir data config cache basedir
XDG_DATA_HOME=$PWD/data XDG_CONFIG_HOME=$PWD/config XDG_CACHE_HOME=$PWD/cache python3 -m qutebrowser -B $PWD/basedir
tree data/ config/ cache/
```

And the `tree` call lists:
```
data/
└── qutebrowser
    └── qutebrowser
        └── QtWebEngine
            └── Default
                └── user_prefs.json
config/
cache/
├── fontconfig
│   ├── 16e13c5e2abffa2743facdd8b290dc70-le64.cache-8
│   ├── 58d6aa9ae16ce6c2bc6f3eb66411357c-le64.cache-8
│   ├── b1521c396d8d77d03639ffc59a54505a-le64.cache-8
│   ├── CACHEDIR.TAG
│   └── f9ea882b4c4e94fab82b09a97ace02dc-le64.cache-8
├── mesa_shader_cache
│   ├── 01
│   │   └── 573f97b5121d1432bc768a9dea1322f45e4c7b
[ ... snip ... ]
│   └── index
├── qtshadercache-x86_64-little_endian-lp64
│   ├── 023b9e4228f85219707672bca89e9a1cf9030480
│   └── 77e560c0c1c4e996635d473a264371cf3612b2e4
└── qutebrowser
    └── qutebrowser
        └── qtpipelinecache-x86_64-little_endian-lp64
```

**Comment by The-Compiler** [comment]
I don't think there is much we can or should do here for the cache data here:

- Almost all of it is from components deep down the stack (fontconfig/mesa). Short of us setting `XDG_CACHE_HOME` to our basedir, I don't think there is a way to influence where those things should go. And if we do that, we will run into fun issues like what to do about subprocesses (#7776).
- The main aim of `--basedir` / `--temp-basedir` is to provide a way to have an isolated instance of qutebrowser not affected by any user-wide data. In theory, this should be the case even if some cache files are present still.

As for `user_prefs.json`, I already reported something very similar to Qt a while ago:

- [[QTBUG-78319] [REG 5.14] user_prefs.json gets written before profile is ready - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-78319)

**Comment by vejkse** [comment]
By looking at backups, I can tell that `~/.local/share/qutebrowser/qutebrowser/QtWebEngine/Default/user_prefs.json` started being created after I update qutebrowser from 2.5.4 to 3.0.0.  This could be due to the change from QtWebEngine 5 to 6?

**Comment by The-Compiler** [comment]
@vejkse Perhaps you have been running Qt < 5.14 before? As outlined in the issue above, this started appearing in Qt 5.14 already, not only with Qt 6.

**Comment by vejkse** [comment]
No, the last version of qt5-webengine in use before that update was 5.15.14.  I have been using a version of Qt ≥ 5.14 since December 2019, without that extra file appearing.

**Comment by The-Compiler** [comment]
No clue then (other than maybe if you never opened the devtools until now). I've been seeing it ever since Qt 5.14. Anyways, doesn't really matter, as it's the Qt bug mentioned above either way.

**Comment by vejkse** [comment]
I don’t think this is related to the devtools, if I delete this `qutebrowser/qutebrowser` directory, it is recreated as soon as I launch qutebrowser with any profile.

**Comment by The-Compiler** [comment]
Heh, that issue now got closed with "Assuming fixed wiht qt6 defaults". Guess this will need some more investigation from qutebrowser's side then...
